Core/Src/main.c

```diff
@@ -134,16 +141,35 @@ void SystemClock_Config(void)
   {
     Error_Handler();
   }
-  PeriphClkInit.PeriphClockSelection = RCC_PERIPHCLK_USART2;
-  PeriphClkInit.Usart2ClockSelection = RCC_USART2CLKSOURCE_PCLK1;
-  if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInit) != HAL_OK)
-  {
-    Error_Handler();
-  }
 }
 
 /* USER CODE BEGIN 4 */
+uint8_t readADC (uint8_t channel){
+	uint8_t controlByte = 0x01;
+	controlByte |= (channel & 0x07) << 4;
+
+	uint8_t receivedData[2] = {0};
+	uint8_t transmitData[3] = {controlByte, 0x00, 0x00};
```

Comment:
set up your transmit buffer as shown by the datasheet. don't merge the start bit and config bits

![image](https://github.com/user-attachments/assets/81b08782-0f6e-4974-8d8e-470369f6340c)